libvmi/read.c

```diff
@@ -97,13 +97,13 @@ vmi_read(
 
         if(dtb) {
             paddr = vmi_pagetable_lookup(vmi, dtb, start_addr + buf_offset);
+            if (!paddr) {
```

Comment:
We just need to thread carefully to remain compatible with both x86 and ARM paging (ENTRY_PRESENT is for x86 only, though I'm sure ARM has something that's equivalent)